apps/nextjs/src/app/[locale]/manage/search-engines/_form.tsx

```diff
@@ -52,11 +58,26 @@ export const SearchEngineForm = (props: SearchEngineFormProps) => {
           </Grid.Col>
         </Grid>
         <IconPicker initialValue={initialValues?.iconUrl} {...form.getInputProps("iconUrl")} />
-        <TextInput
-          {...form.getInputProps("urlTemplate")}
-          withAsterisk
-          label={t("search.engine.field.urlTemplate.label")}
-        />
+        <SegmentedControl data={searchEngineTypes.map((type) => type)} {...form.getInputProps("type")} fullWidth />
+
+        {form.values.type === "generic" && (
+          <TextInput
+            {...form.getInputProps("urlTemplate")}
+            withAsterisk
+            label={t("search.engine.field.urlTemplate.label")}
+          />
+        )}
+
+        {form.values.type === "fromIntegration" && (
+          <WidgetIntegrationSelect
+            label="Integration"
+            data={integrationData ?? []}
+            canSelectMultiple={false}
+            onChange={(value) => form.setFieldValue("integrationId", value[0])}
+            value={form.values.integrationId !== undefined ? [form.values.integrationId] : []}
+          />
+        )}
```

Comment:
Add Fieldset arround Source selection like for integrations:

![image](https://github.com/user-attachments/assets/de4ee4b0-6c06-4410-a84e-52363f020344)

Otherwise it looks like the description would also be part of it